src/test/java/org/folio/innreach/controller/InnReachTransactionControllerTest.java

```diff
@@ -1085,4 +1093,37 @@ private void modifyTransactionState(UUID transactionId, InnReachTransaction.Tran
     repository.save(transaction);
   }
 
+  @ParameterizedTest
+  @ValueSource(strings = {"0aab1720-14b4-4210-9a19-0d0bf1cd64d3",
+                          "ab2393a1-acc4-4849-82ac-8cc0c37339e1",
+                          "79b0a1fb-55be-4e55-9d84-01303aaec1ce"})
+  @Sql(scripts = {
+    "classpath:db/central-server/pre-populate-central-server.sql",
+    "classpath:db/inn-reach-transaction/pre-populate-inn-reach-transaction.sql"
+  })
+  void updateTransactionWhenImmutableFieldsNotChanged(String transactionId) {
+    var transaction = repository.fetchOneById(UUID.fromString(transactionId)).get();
+    var hold = transaction.getHold();
+
+    hold.setFolioRequestId(null);
+    transaction.setHold(hold);
+    transaction.setState(FINAL_CHECKIN);
+
+    var transactionDTO = innReachTransactionMapper.toDTO(transaction);
+
+    var responseEntity = testRestTemplate.exchange(
+      UPDATE_TRANSACTION_ENDPOINT, HttpMethod.PUT,
+      new HttpEntity<>(transactionDTO, headers), InnReachTransactionDTO.class,
+      UUID.fromString(transactionId)
+    );
+
+    var updatedTransaction = repository.fetchOneById(UUID.fromString(transactionId)).get();
+    var updatedState = updatedTransaction.getState();
+    var updatedFolioRequestId = updatedTransaction.getHold().getFolioRequestId();
+
+    assertEquals(HttpStatus.NO_CONTENT, responseEntity.getStatusCode());
```

Comment:
It's also worth to check that immutable field is not modified, according to the requirements

> The following fields on the transaction should be immutable:
> 
>     trackingId
>     centralServerCode
>     type

Reply:
check added